datacube_stats/output_drivers.py

```diff
@@ -560,15 +573,49 @@ def open_output_files(self):
 def _format_filename(path_template, **kwargs):
     x, y = kwargs['tile_index']
     epoch_start, epoch_end = kwargs['time_period']
-    return Path(str(path_template).format(x=x, y=y, epoch_start=epoch_start, epoch_end=epoch_end,
+    percent = kwargs['percent'] 
+    return Path(str(path_template).format(x=x, y=y, epoch_start=epoch_start, epoch_end=epoch_end, percent=percent,
                                           **kwargs))
 
 
 def _polygon_from_sources_extents(sources, geobox):
-    sources_union = geometry.unary_union(source.extent.to_crs(geobox.crs) for source in sources)
+    #sources_union = geometry.unary_union(source.extent.to_crs(geobox.crs) for source in sources)
+    sources_union = unary_union_cus(source.extent.to_crs(geobox.crs) for source in sources)
     valid_data = geobox.extent.intersection(sources_union)
     return valid_data
 
+def _make_geom_from_ogr_cust(geom, crs):
+    result = Geometry.__new__(Geometry)
+    result._geom = geom  # pylint: disable=protected-access
+    result.crs = crs
+    return result
+
+def unary_union_cus(geoms):
```

Comment:
Is this different to the code in `datacube.utils.geometry`? It should be moved to a util module somewhere, or into datacube.